Add Netlify dev reference (#12030)
withastro/docs: src/content/docs/en/guides/integrations-guide/netlify.mdx

});
```

### Dev features

> When running `astro dev`, the adapter enables several Netlify platform features to ensure the envronment matches production as closely as possible. These include:

Reviewer: Yan beat me to this one xD

> When running `astro dev`, the adapter enables several Netlify platform features to ensure the envronment matches production as closely as possible. These include:

Reviewer 2: I gave Yan credit for this one! ;)

- A local [Netlify Image CDN](https://docs.netlify.com/build/image-cdn/overview/#local-development) server. This is used for [images](#netlify-image-cdn-support) by default
- A local [Netlify Blobs](https://docs.netlify.com/build/data-and-storage/netlify-blobs/) server. This is used for [sessions](#sessions) by default

> - [Redirects, rewrites](https://docs.netlify.com/routing/redirects/) and [headers](https://docs.netlify.com/routing/headers/#app) from your Netlify config

Reviewer 3: the links changed a bit (they redirect properly, but it doesn't hurt to update). Also, the `#app` header for the headers link doesn't exist anymore, so I omitted it.
```suggestion
- [Redirects, rewrites](https://docs.netlify.com/manage/routing/redirects/overview/) and [headers](https://docs.netlify.com/manage/routing/headers/) from your Netlify config
```

- Access to [Netlify Edge Context](/en/guides/integrations-guide/netlify/#accessing-edge-context-from-your-site) in on-demand pages
- [Environment variables](https://docs.netlify.com/build/environment-variables/overview/) from your Netlify site

> These work best when your local site is [linked to a Netlify site](https://docs.netlify.com/cli/get-started/#link-and-unlink-sites) using `netlify link`.

Reviewer 2: ```suggestion
These work best when your local site is [linked to a Netlify site](https://docs.netlify.com/api-and-cli-guides/cli-guides/get-started-with-cli/#link-and-unlink-sites) using `netlify link`.
```

You can enable or disable some of these features using the `devFeatures` option in your adapter configuration. By default, all features are enabled except for environment variables.

#### `devFeatures`

<p>

> **Type:** `object | boolean`<br />

Reviewer 2: Great attention to detail! Not sure I would have caught this!

**Default:** `{ images: true, environmentVariables: false }`<br />
<Since v="6.5.1" pkg="@astrojs/netlify"/>
</p>

The `devFeatures` option can be either a boolean to enable or disable all features, or an object to enable specific features.

> - `images`: Enables support for the local [Netlify Image CDN](https://docs.netlify.com/build/image-cdn/overview/#local-development) in development. This uses a local version of the Netlify Image CDN, rather than the default Astro image service. Default: `true`

Reviewer 2: I checked, and this is the pattern we use in the Cloudflare docs, so yes, let's do it!

});
```

### Local development features

When running `astro dev`, the adapter enables several Netlify platform features to ensure the environment matches production as closely as possible. These include:

- A local [Netlify Image CDN](https://docs.netlify.com/build/image-cdn/overview/) server. This is used for [images](#netlify-image-cdn-support) by default.
- A local [Netlify Blobs](https://docs.netlify.com/build/data-and-storage/netlify-blobs/) server. This is used for [sessions](#sessions) by default
- [Redirects, rewrites](https://docs.netlify.com/manage/routing/redirects/overview/) and [headers](https://docs.netlify.com/manage/routing/headers/) from your Netlify config

> - Access to [Netlify Edge Context](/en/guides/integrations-guide/netlify/#accessing-edge-context-from-your-site) in on-demand pages

Reviewer 2: ```suggestion
- Access to [Netlify Edge Context](#accessing-edge-context-from-your-site) in on-demand pages
```
Doesn't make a huge difference, but does make it easier for editors of the docs to know that this is a link within the same page.

- [Environment variables](https://docs.netlify.com/build/environment-variables/overview/) from your Netlify site

These work best when your local site is [linked to a Netlify site](https://docs.netlify.com/api-and-cli-guides/cli-guides/get-started-with-cli/#link-and-unlink-sites) using `netlify link`.

You can enable or disable some of these features using the [`devFeatures`](#devfeatures) option in your adapter configuration. By default, all features are enabled except for environment variables.

#### `devFeatures`

<p>
**Type:** `boolean | object`<br />
**Default:** `{ images: true, environmentVariables: false }`<br />
<Since v="6.5.1" pkg="@astrojs/netlify"/>
</p>

The `devFeatures` option can be either a boolean to enable or disable all features, or an object to enable specific features.

```js title="astro.config.mjs" ins={7-12}
import { defineConfig } from 'astro/config';
import netlify from '@astrojs/netlify';

export default defineConfig({
  // ...
  adapter: netlify({
    devFeatures: {
      // Enable Netlify Image CDN support in dev. Defaults to true.

>       images: true, 

Reviewer 2: ```suggestion
      images: false, 
```
We typically wouldn't show setting to true, since that's the default and no one would need to do this.

But this brings up a good point: if they *do* configure an object, *must* they specify values for both (because this overrides), or do the defaults work when not all properties are specified?

- [Environment variables](https://docs.netlify.com/build/environment-variables/overview/) from your Netlify site

These work best when your local site is [linked to a Netlify site](https://docs.netlify.com/api-and-cli-guides/cli-guides/get-started-with-cli/#link-and-unlink-sites) using `netlify link`.

You can enable or disable some of these features using the [`devFeatures`](#devfeatures) option in your adapter configuration. By default, all features are enabled except for environment variables.

#### `devFeatures`

<p>
**Type:** `boolean | object`<br />
**Default:** `{ images: true, environmentVariables: false }`<br />
<Since v="6.5.1" pkg="@astrojs/netlify"/>
</p>

The `devFeatures` option can be either a boolean to enable or disable all features, or an object to enable specific features.

##### `images`

<p>
**Type:** `boolean`<br />
**Default:** `true`<br />
<Since v="6.5.1" pkg="@astrojs/netlify"/>
</p>

Enables support for the local [Netlify Image CDN](https://docs.netlify.com/build/image-cdn/overview/) in development.

This uses a local version of the Netlify Image CDN, rather than the default Astro image service. 

> ##### `environmentVariables``

Reviewer 2: ```suggestion

##### `devFeatures.environmentVariables``
```

<p>
**Type:** `boolean`<br />
**Default:** `false`<br />
<Since v="6.5.1" pkg="@astrojs/netlify"/>

> </p>`

Reviewer 3: ```suggestion
</p>
```
